illumos-utils/src/zfs.rs

```diff
@@ -181,6 +201,20 @@ impl Zfs {
         Ok(filesystems)
     }
 
+    /// Return the name of a dataset for a ZFS object.
+    ///
+    /// The object can either be a dataset name, or a path, in which case it
+    /// will be resolved to the _mounted_ ZFS dataset containing that path.
+    pub fn get_dataset_name(object: &str) -> Result<String, ListDatasetsError> {
+        let mut command = std::process::Command::new(ZFS);
+        let cmd = command.args(&["get", "-Hpo", "name", "name", object]);
```

Comment:
It actually doesn't make any difference, but I think you technically want the `value` column for the `name` property.
```suggestion
        let cmd = command.args(&["get", "-Hpo", "value", "name", object]);
```

Reply:
Nice catch, done in c2c6ed4b2